File: program/program/find-the-transpose-of-a-matrix/find_the_transpose_of_a_matrix.py
matrix = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]

rows = len(matrix)
columns = len(matrix[0])

transposed_matrix = [[0 for x in range(rows)] for y in range(columns)]

for i in range(columns):
    for j in range(rows):
        transposed_matrix[i][j] = matrix[j][i]

for i in range(columns):
    for j in range(rows):
        print(transposed_matrix[i][j], " ", end="")
    print()

File: program/program/find-the-transpose-of-a-matrix/FindTheTransposeOfAMatrix.swift
func transpose(matrix: [[Int]]) -> [[Int]] {
    // Guard against empty matrix
    guard !matrix.isEmpty else { return [[]] }

    let rowCount = matrix.count
    let colCount = matrix[0].count

    // Initialize an empty matrix for the transpose
    var transposedMatrix = Array(repeating: Array(repeating: 0, count: rowCount), count: colCount)

    // Iterate through the matrix and swap rows with columns
    for row in 0..<rowCount {
        for col in 0..<colCount {
            transposedMatrix[col][row] = matrix[row][col]
        }
    }

    return transposedMatrix
}

// Example usage
let inputMatrix = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
]

let outputMatrix = transpose(matrix: inputMatrix)
print("Output:")
for row in outputMatrix {
    print(row.map { String($0) }.joined(separator: " "))
}
